test/e2e/common.go

```diff
@@ -7,63 +7,94 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// newPod returns a new Pod object.
-func newPod(namespace string, name string, containerName string, runtimeclass string) *corev1.Pod {
-	return &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
-		Spec: corev1.PodSpec{
-			Containers:       []corev1.Container{{Name: containerName, Image: "nginx"}},
-			DNSPolicy:        "ClusterFirst",
-			RestartPolicy:    "Never",
-			RuntimeClassName: &runtimeclass,
-		},
+type RuntimeClass string
+
+const (
+	RuntimeClassKata       RuntimeClass = "kata"
+	RuntimeClassKataCLH    RuntimeClass = "kata-clh"
+	RuntimeClassQemu       RuntimeClass = "kata-qeum"
+	RuntimeClassKataRemote RuntimeClass = "kata-remote"
+)
+
+func (rc *RuntimeClass) String() string {
+	switch *rc {
+	case RuntimeClassKata:
+		return "kata"
+	case RuntimeClassKataCLH:
+		return "kata-clh"
+	case RuntimeClassQemu:
+		return "kata-qeum"
+	case RuntimeClassKataRemote:
+		return "kata-remote"
+	default:
+		panic("Unknown runtime class")
 	}
 }
 
+type PodConfigMapBinding struct {
+	mountPath     string
+	configMapName string
+}
+
+type PodSecretBinding struct {
+	mountPath  string
+	secretName string
+}
+
 // newPod returns a new Pod object.
-func newBusyboxPod(namespace string, name string, containerName string, runtimeclass string) *corev1.Pod {
-	return &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
-		Spec: corev1.PodSpec{
-			Containers:       []corev1.Container{{Name: containerName, Image: "quay.io/prometheus/busybox:latest", Command: []string{"/bin/sh", "-c", "sleep 3600"}}},
-			DNSPolicy:        "ClusterFirst",
-			RuntimeClassName: &runtimeclass,
-			RestartPolicy:    corev1.RestartPolicyAlways,
-		},
+func newPod(namespace string, name string, containerName string, runtimeClass RuntimeClass, imageName string, restartPolicy corev1.RestartPolicy, command []string, podConfigMapBinding *PodConfigMapBinding, podSecretBinding *PodSecretBinding) *corev1.Pod {
```

Comment:
So one of the things that I was thinking for this was either the [Options Pattern](https://levelup.gitconnected.com/options-pattern-in-golang-9a0384a9d8db#:~:text=Option%20pattern%20is%20a%20functional,use%20it%20when%20it%20is.), or maybe [Builder Pattern](https://medium.com/codex/builder-pattern-in-go-15b400fd4b1e).

This would allow us to make this function more user friendly by not having to pass in all the fields and then grow it every time there is a new parameter we'd want which would involve updating every invocation of it, so would be more flexible, so I think those approaches are worth considering

Reply:
Hi @mattarnoatibm , I started the review but realized you implemented Steve's suggestion on a separated commit. Could you squash the three commits? It will help on the review.

Reply:
Hi @wainersm . I've squashed the 3 commits into 1. Thanks for starting the review for this PR.